### PR TITLE
Astro/download page [RFR]

### PR DESCRIPTION
**Pull request by djacu**
To reviewers:
Please check that the NixOS AWS EC2 selection works correctly and updates the table.

- [x] Fix vertical centering on the tabs.
- [x] Panels that aren't the full width are not filling the remaining column.
- [x] Need to add mobil (small screen) styling.
- [x] Need to extract AMIs into a content file and map over it.
- [x] Fix tabs not filling the parent.
- [ ] Tabs don't look exactly the same as the original when the screen is small (when the cross over to medium size).

### CODE REVIEW

**Comment by djacu** [comment]
@garbas @thilobillerbeck This is ready for review. It works as far as I can tell.

One thing I didn't realize was how the tabs for the different section changed when the screen is less wide. I didn't account for this when designing the components. It can probably be fixed but I haven't figured out how to do it yet.

You can see the comparison below. The old way hides the original tabs and unhides these header tags that are used to collapse the content after. The way I have it now is that the just changes the grid layout.

What do you think? Do we want to match the old way?
If so, and either of you want to give it a try, please feel free.
I've been working on this for 3 days and need a break.

<details><summary>OLD</summary>

image
</details>

<details><summary>NEW</summary>

image
</details>

**Review by thilobillerbeck — CHANGES_REQUESTED** [review]
Looks good, thanks for your work! There are some small things I marked with a comment but nothing too big.

One thing this implementation clearly leaves open is how we deal with hard coded links to things like NixOS releases, I think it would be better if there would be a place to just swap verison numbers on release and all URLS adapt.

Apart from that I like the mobile UI. It is (against what we planned) not a 1:1 port of the original design but it's okay for me. In the future we maybe could add some kind of dropdown at the top with a heading like `Choose platform` to save on mobile screen space, but it's okay for now. :)

File: src/components/pages/download/components/recommended.astro
[the remark this answers is not in the document]
Comment (djacu) [reply src/components/pages/download/components/recommended.astro:1]: Done!

File: src/components/pages/download/components/DownloadButtonAndSHA.astro
[the remark this answers is not in the document]
Comment (djacu) [reply src/components/pages/download/components/DownloadButtonAndSHA.astro:1]: Done!

**Comment by djacu** [comment]
@thilobillerbeck all changes resolved
See https://github.com/NixOS/nixos-homepage/pull/1164/commits/2a2b082048cb8fd1b109de65e24c4eb328520725 for modifications to the Button class to allow for optional SHA arguments.
See https://github.com/NixOS/nixos-homepage/pull/1164/commits/86e3ee3801ca73138b6b48f8a2ee475048342d8e for the new Tag class.

**Review by thilobillerbeck — APPROVED** [review]

**Comment by thilobillerbeck** [comment]
looks good, the Button on the `/learn` page seems to be broken but this is a fix we can incorporate shortly before publishing the site. Thx for the work.

**Pull request merged**